verifier/src/lib.rs

```diff
@@ -188,7 +188,9 @@ where
     // read the out-of-domain trace frames (the main trace frame and auxiliary trace frame, if
     // provided) sent by the prover and evaluate constraints over them; also, reseed the public
     // coin with the OOD frames received from the prover.
-    let (ood_main_trace_frame, ood_aux_trace_frame) = channel.read_ood_trace_frame();
+    let ood_trace_frame = channel.read_ood_trace_frame();
+    let ood_main_trace_frame = ood_trace_frame.main_frame();
+    let ood_aux_trace_frame = ood_trace_frame.aux_frame();
```

Comment:
minor nit: we probably don't need to introduce variables for these and make calls directly in the lines below like so:
```Rust
let ood_constraint_evaluation_1 = evaluate_constraints(
    &air,
    constraint_coeffs,
    &ood_trace_frame.main_frame(),
    &ood_trace_frame.aux_frame(),
    aux_trace_rand_elements,
    z,
);
```

Reply:
Ah! Then let's keep it the way you have it.